### PR TITLE
Small improvement to tethered boot support from macOS host

### PR DESCRIPTION
**Pull request by aderuelle**
Small improvement to the out-of-the-box experience when using maOS as tethered boot host

- use `/dev/cu.usbmodemP_03` and `/dev/cu.usbmodemP_01` for `picocom-sec.sh` and `proxyclient` respectively

### CODE REVIEW

**Comment by marcan** [comment]
LGTM, feel free to unmark as draft.

**Comment by martell** [comment]
can confirm this works for me out of the box on both a macos and linux host, thank you @aderuelle I carry this on my local envs

File: proxyclient/m1n1/proxy.py [hunk 1]
```diff
@@ -131,13 +131,20 @@ class UartInterface(Reloadable):
     REPLY_LEN = 36
     EVENT_HDR_LEN = 8
 
+    DEFAULT_UART_DEV="/dev/m1n1"
+    DEFAULT_BAUD_RATE="115200"
```
Comment (marcan) [inline proxyclient/m1n1/proxy.py:135]: This needs to be an int

File: proxyclient/tools/picocom-sec.sh [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+#
+# Secondary UART device name exposed when m1n1 boots is /dev/m1n1-sec on
+# Linux host and /dev/cu.usbmodemP_03 on macOS host
+#
+DEFAULT_SECDEV=/dev/m1n1-sec
+PLATFORM_NAME=$(uname)
+
+case "${PLATFORM_NAME}" in
+Linux)
+    SECDEV=/dev/ttyACM1 ;;
```
Comment (marcan) [inline proxyclient/tools/picocom-sec.sh:11]: Same here, the udev rules make `/dev/m1n1-sec` work on Linux (more reliably than guessing ACM device numbers).

File: proxyclient/m1n1/proxy.py [hunk 3]
```diff
@@ -131,13 +131,20 @@ class UartInterface(Reloadable):
...
+    if platform.system() == 'Linux':
+        DEFAULT_UART_DEV="/dev/ttyACM0"
```
Comment (marcan) [inline proxyclient/m1n1/proxy.py:137]: This changes the existing behavior. The idea is that on Linux you install the udev rules in `udev/`, which gives you the `/dev/m1n1` symlink. Maybe we can just leave Linux at the default?

**Review by marcan — CHANGES_REQUESTED** [review]

File: proxyclient/m1n1/proxy.py (hunk 1 above)
Comment (aderuelle) [reply proxyclient/m1n1/proxy.py:135]: Yeah sure

File: proxyclient/m1n1/proxy.py (hunk 3 above)
Comment (aderuelle) [reply proxyclient/m1n1/proxy.py:137]: Yes fine with me to keep Linux as default

File: proxyclient/tools/picocom-sec.sh (hunk 2 above)
Comment (aderuelle) [reply proxyclient/tools/picocom-sec.sh:11]: Nice, I should have figured this out … I will remove this and add some comments about the udev stuff

**Comment by aderuelle** [comment]
> can confirm this works for me out of the box on both a macos and linux host thank you @aderuelle, I carry this on my local envs

thx @martell I removed Linux raw device names as follow up to review comment. 

On Linux this is handled by the[ provided udev rule](https://github.com/AsahiLinux/m1n1/blob/main/udev/80-m1n1.rules) whereas on macOS this is currently hardcoded (there's no udev-like framework AFAIK).

However, we could poll the ioreg database, I've write a [small Python script to do just that.](https://gist.github.com/aderuelle/36454ed691dfceba575411721d8d18b8) I'll have a look into plugging it in someday.

**Pull request merged**